gbc/src/stupid_gfx_joypad_provider.rs

```diff
@@ -0,0 +1,36 @@
+extern crate stupid_gfx;
+extern crate lib_gbc;
+
+use lib_gbc::machine::{
+    joypad_provider::JoypadProvider,
+    joypad::Joypad
+};
+use stupid_gfx::{
+    event_handler::EventHandler,
+    event::{Event,Scancode}
+};
+
+pub struct StupidGfxJoypadProvider<'a>{
+    event_handler:&'a mut EventHandler
+}
+
+impl<'a> StupidGfxJoypadProvider<'a>{
+    pub fn new(handler:&'a mut EventHandler)->Self{
+        StupidGfxJoypadProvider{
+            event_handler:handler
+        }
+    }
+}
+
+impl<'a> JoypadProvider for StupidGfxJoypadProvider<'a>{
+    fn provide(&mut self, joypad:&mut Joypad) {
+        joypad.a        = self.event_handler.has_event_occurred(Event::KeyPressed(Scancode::X));
```

Comment:
Maybe (if this could be initialized once create an enum for the buttons and than map those from outside the struct.